plat/kvm/arm/link64.lds.S

```diff
@@ -42,6 +42,16 @@ OUTPUT_FORMAT("elf64-littleaarch64")
 OUTPUT_ARCH(aarch64)
 ENTRY(_libkvmplat_entry)
 
+PHDRS
+{
+	headers PT_PHDR PHDRS;
+	text PT_LOAD FILEHDR PHDRS FLAGS(5);
```

Comment:
I have a question here. When you build a unikernel you get something like this:
```
Program Headers:
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  PHDR           0x0000000000000040 0x00000000000ff040 0x00000000000ff040
                 0x0000000000000150 0x0000000000000150  R      0x8
  LOAD           0x0000000000000000 0x00000000000ff000 0x00000000000ff000
                 0x00000000002467ce 0x00000000002467ce  R E    0x1000
  LOAD           0x0000000000247000 0x0000000000346000 0x0000000000346000
                 0x00000000000760c8 0x00000000000760c8  R      0x1000
  LOAD           0x00000000002be000 0x00000000003bd000 0x00000000003bd000
                 0x00000000001ad3a8 0x00000000001f37a8  RW     0x1000
  TLS            0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000000000000         0x8
  GNU_STACK      0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000000000000  RW     0x8
```
The headers are located in the first page (@`0xff000`) of the first `LOAD` segment. So they share the **executable** permissions of the `.text` segment. Means you could execute whatever data comprises the ELF header 😱 

Now it is alright with multiboot because we do not load that part. However, this might be different for the ASLR case, where we would use the ELF headers to load the kernel.

Reply:
I moved the headers into their own segment.

Reply:
> I replaced the objcopy step by making the ELF loadable using multiboot directly. Thanks @mkroening, for the hints regarding multiboot ELF loading.

I just saw this change. Interesting! You state that it is loadable by grub in the commit message. It is still loadable by QEMU directly, too?